packages/dds/merge-tree/src/opBuilder.ts

```diff
@@ -31,7 +32,10 @@ export function createAnnotateMarkerOp(
 	if (!id) {
 		return undefined;
 	}
-
+	assert(
+		!props.markerId || props.markerId === id,
+		"Cannot change the markerId of an existing marker",
+	);
```

Comment:
If you annotate a range (rather than a marker) and the range contains a marker, will it modify the markerId property? I'm not sure it's a valid usage, but it might bypass this check. In which case, it would be better to check in MergeTree.annotateRange's annotateSegment.

Reply:
The marker is a segment. You can call Marker.is on it and get the id.